server/routes/legal.tsx

```diff
@@ -15,34 +15,52 @@ app.get('/privacy/settings', (_, res) => res.redirect('/legal/settings'));
 app.get('/tos', (_, res) => res.redirect('/legal/terms'));
 app.get('/legal', (_, res) => res.redirect('/legal/terms'));
 
-app.get('/legal/:tab', (req, res, next) => {
+const tabToTitle = {
+	terms: 'Terms of Service',
+	privacy: 'Privacy Policy',
+	aup: 'Acceptable Use Policy',
+	settings: 'Privacy & Account',
+};
+
+app.get('/legal/:tab', async (req, res, next) => {
 	if (!['terms', 'privacy', 'aup', 'settings'].includes(req.params.tab)) {
 		return next();
 	}
-	const userId = req.user?.id;
-	return Promise.all([
-		getInitialData(req),
-		userId ? getIntegrations(userId) : [],
-		userId ? getOrCreateUserNotificationPreferences(userId) : undefined,
-	])
-		.then(([initialData, integrations, userNotificationPreferences]) => {
-			return renderToNodeStream(
-				res,
-				// @ts-expect-error ts-migrate(2322) FIXME: Type '{ children: Element; chunkName: string; init... Remove this comment to see the full error message
-				<Html
-					chunkName="Legal"
-					initialData={initialData}
-					viewData={{ integrations, userNotificationPreferences }}
-					headerComponents={generateMetaComponents({
-						initialData,
-						title: `Legal · ${initialData.communityData.title}`,
-						description: initialData.communityData.description,
-					})}
-				>
-					{/* @ts-expect-error ts-migrate(2322) FIXME: Type '{ communityData: { title: string; descriptio... Remove this comment to see the full error message */}
-					<Legal {...initialData} />
-				</Html>,
-			);
-		})
-		.catch(handleErrors(req, res, next));
+
+	try {
+		const userId = req.user?.id;
+
+		const [initialData, integrations, userNotificationPreferences] = await Promise.all([
+			getInitialData(req),
+			userId ? getIntegrations(userId) : [],
+			userId ? getOrCreateUserNotificationPreferences(userId) : undefined,
+		]);
+
+		const title = tabToTitle[req.params.tab as keyof typeof tabToTitle];
+		const isSettingsTab = req.params.tab === 'settings';
+
+		return renderToNodeStream(
+			res,
+			// @ts-expect-error ts-migrate(2322) FIXME: Type '{ children: Element; chunkName: string; init... Remove this comment to see the full error message
+			<Html
+				chunkName="Legal"
+				initialData={initialData}
+				viewData={{
+					integrations,
+					userNotificationPreferences,
+					userEmail: isSettingsTab ? req.user?.email : undefined,
```

Comment:
I want to show what the user's current email is on the page, hence why im passing it through like this